File: Prosary/Mocks/Content/MysteryTranslations.swift
//
//  MysteryTranslations.swift
//  Prosary
//
//  Looks up the title/fruit/description of a mystery by imageKey and language code, falling
//  back to Latin when a translation is missing. See MysteryTranslations+*.swift for the actual
//  per-language tables.
//

import Foundation

enum MysteryTranslations {
    static func get(languageCode: String?, imageKey: String) -> MysteryText {
        if let languageCode, let table = byLanguage[languageCode], let text = table[imageKey] {
            return text
        }

        return latin[imageKey] ?? MysteryText(title: imageKey, fruit: "", description: "")
    }

    static let byLanguage: [String: [String: MysteryText]] = [
        "la": latin,
        "en": english,
        "ar": arabic,
        "he": hebrew,
    ]
}

File: app/src/main/java/com/dkaluta/Prosary/content/MysteryTranslations.kt
package com.dkaluta.Prosary.content

/** Looks up the title/fruit/description of a mystery by imageKey and language code, falling
 * back to Latin when a translation is missing. See MysteryTranslations{Language}.kt for the
 * actual per-language tables. */
object MysteryTranslations {
    fun get(languageCode: String?, imageKey: String): MysteryText {
        val table = languageCode?.let { byLanguage[it] }
        val text = table?.get(imageKey)
        if (text != null) return text

        return mysteryTranslationsLatin[imageKey] ?: MysteryText(title = imageKey, fruit = "", description = "")
    }

    val byLanguage: Map<String, Map<String, MysteryText>> = mapOf(
        "la" to mysteryTranslationsLatin,
        "en" to mysteryTranslationsEnglish,
        "ar" to mysteryTranslationsArabic,
        "he" to mysteryTranslationsHebrew,
        "ru" to mysteryTranslationsRussian,
        "tl" to mysteryTranslationsTagalog,
    )
}
